### PR TITLE
[chore] internal/ap: add pollable AS types, code reformatting, general niceties

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/ap/activitystreams.go [hunk 1]
```diff
@@ -78,3 +78,49 @@ const (
...
+		ActivityFlag,
+		ActivityDislike:
+		return true
+	default:
+		return false
+	}
+}
+
+// isIntransitiveActivity returns whether AS type name is of an IntransitiveActivity.
+func isIntransitiveActivity(typeName string) bool {
```
Comment (daenney) [inline internal/ap/activitystreams.go:117]: Can this become `return !isActivity` instead? Or is there stuff in between that doesn't fall into either category?

Comment (NyaaaWhatsUpDoc) [reply internal/ap/activitystreams.go:117]: normally i'd agree, but given how absurdly flexible ActivityStreams vocab can be i think it's worth specifically testing for types they mark as extending `IntransitiveActivity`

File: internal/ap/extract.go [hunk 2]
```diff
@@ -22,18 +22,60 @@ import (
...
 	"github.com/superseriousbusiness/activity/pub"
+	"github.com/superseriousbusiness/activity/streams/vocab"
 	"github.com/superseriousbusiness/gotosocial/internal/gtserror"
 	"github.com/superseriousbusiness/gotosocial/internal/gtsmodel"
 	"github.com/superseriousbusiness/gotosocial/internal/text"
 	"github.com/superseriousbusiness/gotosocial/internal/util"
 )
 
+// ExtractObject will extract an object vocab.Type from given implementing interface.
+func ExtractObject(with WithObject) vocab.Type {
```
Comment (tsmethurst) [inline internal/ap/extract.go:39]: We'll have to be wary using this, because it will return nil (i think) for object entries that are just a URL.

Comment (NyaaaWhatsUpDoc) [reply internal/ap/extract.go:39]: Yes I was thinking of adding a matching `ExtractObjectIRI()` function that can be used for when this returns nil

**Review by tsmethurst — APPROVED** [review]
lgtm! :rainbow:

**Pull request merged**